test/peer_SUITE.erl

```diff
@@ -155,4 +156,19 @@ association_test(_) ->
 
     ?assertError(invalid_association_signature, libp2p_peer:verify(InvalidPeer)),
 
-    ok.
\ No newline at end of file
+    ok.
+
+signed_metadata_test(_) ->
+    #{secret := PrivKey1, public := PubKey1} = libp2p_crypto:generate_keys(ecc_compact),
+    SigFun1 = libp2p_crypto:mk_sig_fun(PrivKey1),
+
+    PeerMap = #{pubkey => libp2p_crypto:pubkey_to_bin(PubKey1),
+                listen_addrs => ["/ip4/8.8.8.8/tcp/1234"],
+                connected => [],
+                nat_type => static,
+                signed_metadata => #{<<"hello">> => <<"world">>}},
+    Peer = libp2p_peer:from_map(PeerMap, SigFun1),
+    ?assertEqual(<<"world">>, libp2p_peer:signed_metadata_get(Peer, <<"hello">>, <<"dlrow">>)),
```

Comment:
you're not encoding and decoding the created peer to verify that the proto definition works

Reply:
Yeah, I will do that.

Reply:
This has been done.